src/plugins/intel_cpu/src/cpu_shape.cpp

```diff
@@ -49,5 +49,29 @@ std::string Shape::toString() const  {
     return output.str();
 }
 
+Shape mergeShapes(const Shape& lhs, const Shape& rhs) {
+    OPENVINO_ASSERT(lhs.getRank() == rhs.getRank(),
+        "Couldn't merge shapes of different ranks: shape 1:",
+        lhs.toString(),
+        " shape 2: ",
+        rhs.toString());
+
+    const auto& lhsMinDims = lhs.getMinDims();
+    const auto& lhsMaxDims = lhs.getMaxDims();
+    const auto& rhsMinDims = rhs.getMinDims();
+    const auto& rhsMaxDims = rhs.getMaxDims();
+
+    VectorDims resultMinDims(lhsMinDims.size());
+    VectorDims resultMaxDims(lhsMaxDims.size());
+
+    for (size_t i = 0; i < resultMinDims.size(); ++i) {
+        OPENVINO_ASSERT(lhsMinDims[i] <= rhsMaxDims[i], "Couldn't merge shapes as the dims intervals are not overlapping.");
+        OPENVINO_ASSERT(rhsMinDims[i] <= lhsMaxDims[i], "Couldn't merge shapes as the dims intervals are not overlapping.");
```

Comment:
I think do this check `resultMaxDims[i] < resultMinDims`, after merge can replace these two.

Reply:
Done. Thank you!